src/lib/containers/download_list/DownloadListTable.tsx

```diff
@@ -204,28 +218,99 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
   }
 
+  const sortColumn = async (column: string) => {
+    try {
+      setIsLoading(true)
+
+      setColumns({
+        file: column === 'file' ? !columns[column] : false,
+        createdBy: column === 'createdBy' ? !columns[column] : false,
+        createdOn: column === 'createdOn' ? !columns[column] : false,
+        size: column === 'size' ? !columns[column] : false,
+      })
+
+      const filesToDownload = downloadList?.filesToDownload ?? []
+
+      filesToDownload.sort((itemA, itemB) => {
+        return sortDownLoadList(itemA, itemB, column)
+      })
+      setData({
+        ...data,
+        downloadList,
+      })
+      listUpdatedCallback?.()
+    } catch (err) {
+      console.error(err)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
   const sortDownLoadList = (
     itemA: FileHandleAssociation,
     itemB: FileHandleAssociation,
+    column: string,
   ) => {
-    let fileNameA = ''
-    let fileNameB = ''
+    let fileName_A: string | undefined = ''
+    let fileName_B: string | undefined = ''
+
+    let createdBy_A: string | undefined = ''
+    let createdBy_B: string | undefined = ''
+    let createdOn_A: string | undefined = ''
+    let createdOn_B: string | undefined = ''
+    let contentSize_A: number | undefined = undefined
+    let contentSize_B: number | undefined = undefined
 
     const fileHandleId = itemA ? itemA.fileHandleId : itemB.fileHandleId
 
     const fileResult = requestedFiles.find(
       fileRes => fileRes.fileHandleId === fileHandleId,
     )
-
     const fileHandle = fileResult ? fileResult.fileHandle : undefined
 
+    // fileHandle is defined, this file is downloadable, show its metadata
     if (fileHandle && itemA) {
-      fileNameA = fileHandle.fileName
+      fileName_A = fileHandle.fileName
+      createdBy_A = fileHandle.createdBy
+      createdOn_A = fileHandle.createdOn
```

Comment:
I think there's a bug here, right now this code only look's at the detailed information for itemA, but not itemB. I would refactor to pull out the information for both using a helper function -

```tsx
const getFileHandleInfo = (
    item: FileHandleAssociation,
) => {
    const fileResult = requestedFiles.find(
      fileRes => fileRes.fileHandleId === item.fileHandleId,
    )
    let fileName = ''
    let createdBy = ''
    let createdOn_A = ''
    let contentSize = ''
    if (fileHandle && item) {
        fileName = fileHandle.fileName
        createdBy = fileHandle.createdBy
        createdOn = fileHandle.createdOn
        contentSize =fileHandle.contentSize
    } else {
       // file is not downloadable, only show its name from entity header info
      const requestedFile = results.find(req => req.id === item.associateObjectId)
      if (requestedFile) {
        fileName = requestedFile?.name
        createdBy = requestedFile?.createdBy
        createdOn = requestedFile?.createdOn
     }
    }
   return { fileName, createdBy,  createdOn,  contentSize }
}

const { fileName: fileName_A, createdBy: createdBy_A,  createdOn: createdOn_A,  contentSize: contentSize_A } = getFileHandleInfo(itemA)
const { fileName: fileName_B, createdBy: createdBy_B,  createdOn: createdOn_B,  contentSize: contentSize_B } = getFileHandleInfo(itemB)

```